lib/browser/tag/update.js

```diff
@@ -163,7 +163,7 @@ export function updateExpression(expr) {
   }
 
   // remove original attribute
-  if (expr.attr && (!expr.wasParsedOnce || !hasValue || value === false)) {
+  if (expr.attr && (!expr.wasParsedOnce || (!hasValue && (dom.tagName !=='INPUT' && dom.tagName !=='OPTION')) || value === false)) {
```

Comment:
I think here we should target only the `input[type='checkbox'],input[type='radio']` and not all of them

Reply:
Dunno what vue does.

The empty string, though, is a perfectly valid value for **all** the input types.

Setting the `input`s value to empty string SHOULD NOT in any way remove the `value` attribute

Reply:
@sourcegr I think we can enable this patch via [`riot.settings`](https://riot.js.org/api/misc/) avoiding breaking changes and major release bump. Per default this patch will be disabled